packages/authorization-process/src/services/authorizationService.ts

```diff
@@ -966,18 +967,89 @@ export function authorizationServiceBuilder(
         )
       );
     },
+    async createProducerKeychainKeys({
+      producerKeychainId,
+      authData,
+      keysSeeds,
+      correlationId,
+      logger,
+    }: {
+      producerKeychainId: ProducerKeychainId;
+      authData: AuthData;
+      keysSeeds: authorizationApi.KeysSeed;
+      correlationId: string;
+      logger: Logger;
+    }): Promise<ProducerKeychain> {
+      logger.info(`Creating keys for producer keychain ${producerKeychainId}`);
+      const producerKeychain = await retrieveProducerKeychain(
+        producerKeychainId,
+        readModelService
+      );
+      assertOrganizationIsProducerKeychainProducer(
+        unsafeBrandId(authData.organizationId),
+        producerKeychain.data
+      );
+      assertProducerKeychainKeysCountIsBelowThreshold(
+        producerKeychainId,
+        producerKeychain.data.keys.length + keysSeeds.length
+      );
+
+      if (!producerKeychain.data.users.includes(authData.userId)) {
+        throw userNotFound(authData.userId, authData.selfcareId);
+      }
+
+      await assertUserSelfcareSecurityPrivileges({
+        selfcareId: authData.selfcareId,
+        requesterUserId: authData.userId,
+        consumerId: authData.organizationId,
+        selfcareV2InstitutionClient,
+        userIdToCheck: authData.userId,
+      });
+
+      if (keysSeeds.length !== 1) {
```

Comment:
since this is a new endpoint, let's define in the spec that this field is not an array but a single object.
(This change will be done as well for clients, once the BFF will be migrated)

Reply:
This also required a change in the [openapi update PR](https://github.com/pagopa/interop-be-monorepo/pull/807), you can see the commit [here](https://github.com/pagopa/interop-be-monorepo/pull/807/commits/680a25711d10473303f7a52fa536618b5899df1b)